src/main/scala/shine/OpenCL/Compilation/SeparateHostAndKernelCode.scala

```diff
@@ -59,7 +60,7 @@ object SeparateHostAndKernelCode {
       freeNats match {
         case v +: rest => iterNats(
           DepLambda[NatKind](NatIdentifier(v.name, v.range))(definition),
-          Natural(v) +: args, rest)
+          Literal(NatAsIntData(v)) +: args, rest)
```

Comment:
so now the argument is an `int` instead of a `nat`, which is in the end going to be the same at runtime, but why this change?